### PR TITLE
Remove macro and update makefile

### PR DESCRIPTION
**Pull request by CollieIsCute**

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -79,11 +80,11 @@ int main(int argc, char* argv[]) {
 
 /* handlers */
 
-#define invalid(k, v)                \
-	{                                \
-		templateSet(template, k, v); \
-		valid = false;               \
-	}
+static int invalid(Template* template, char* key, char* value) {
```
Comment (25077667) [inline main.c:83]: The key and value should be immutable, right?

File: Makefile [hunk 2]
```diff
@@ -50,7 +50,7 @@ tests/%.o: tests/%.c $(OBJS)
 
 test: $(TEST_UNIT_OBJ)
 	@echo Do testing...
-	@python3 tests/driver.py
+	@tests/driver.py
```
Comment (25077667) [inline Makefile:53]: The alpine's /usr/bin/python is a soft link to python2.
Therefore, we should specify the python with version 3.

**Review by 25077667 — COMMENTED** [review]
Please check the comments. Thanks.

File: Makefile (hunk 2 above)
Comment (CollieIsCute) [reply Makefile:53]: 可是你 driver.py 裡面不是已經有 Shebang 了嗎？我覺得已經有寫就不用再一次了。

Comment (CollieIsCute) [reply Makefile:53]: - [shebang 如果 python 版本不符參考資料](https://stackoverflow.com/questions/66287443/python-what-is-the-shebang-line-supposed-to-do-if-the-stated-version-is-not-in)

File: main.c (hunk 1 above)
Comment (CollieIsCute) [reply main.c]: Fixed. Pls chek again.

**Comment by CollieIsCute** [comment]
# All problems are fixed. 
- [O] const problems
- [O] shebang issue

**Review by 25077667 — COMMENTED** [review]
Thanks.

**Pull request merged**